Source/JavaScriptCore/jit/AssemblyHelpers.h

```diff
@@ -980,6 +980,22 @@ class AssemblyHelpers : public MacroAssembler {
 #endif
     }
 
+#if USE(JSVALUE64)
+    void toBigInt64(GPRReg cellGPR, GPRReg destGPR, GPRReg scratchGPR, GPRReg scratch2GPR)
+    {
+        ASSERT(noOverlap(cellGPR, destGPR, scratchGPR, scratch2GPR));
+        load32(Address(cellGPR, JSBigInt::offsetOfLength()), destGPR);
+        JumpList doneCases;
+        doneCases.append(branchTest32(Zero, destGPR));
+        loadPtr(Address(cellGPR, JSBigInt::offsetOfData()), scratchGPR);
+        cageConditionallyAndUntag(Gigacage::Primitive, scratchGPR, destGPR, scratch2GPR, false, false);
+        load64(Address(scratchGPR), destGPR);
```

Comment:
Just curious, how do we handle overflows?

Reply:
only the low 64bit will be used. For negative value, two's complement will be used.